server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRunnable.java

```diff
@@ -81,10 +81,9 @@ public DeliveryRunnable(MailQueue queue, RemoteDeliveryConfiguration configurati
 
     public void start() {
         disposable = Flux.from(queue.deQueue())
-            .publishOn(Schedulers.newParallel("RemoteDelivery", configuration.getWorkersThreadCount()))
+            .publishOn(Schedulers.newElastic("RemoteDelivery", configuration.getWorkersThreadCount()))
```

Comment:
going from a fixed thread pool to an elastic one is probably a good idea because we do mainly IO here but it changes the semantic for workers thread count configuration.
Does it even make sense to keep this configuration?